drud/ddev
# Improve import-db/export-db with --target-db and --no-drop flags, fixes #1961, fixes #1652

## The Problem/Issue/Bug:

#1961 requested the ability (later the default) of dropping tables
#1652 requested an additional flag to target a secondary database

## How this PR Solves The Problem:

* Goes back to import-db dropping all tables as the default
* Adds --no-drop flag for when that's not what's wanted
* Adds --target-db to target another database (and create it if necessary)
* Minor docs added to the FAQ section about working with additional databases
* Adds optional "project" arg to both import-db and export-db
* Adds --target-db also to `ddev export-db`

## Manual Testing Instructions:

`ddev export-db <project> >someproject.sql.gz`
`ddev import-db <someproject.sql.gz` 
`ddev import-db --target-db newdb --src someproject.zql.gz`, verify contents of "newdb"
`ddev import-db` another smaller one. Only the new tables should still exist.
`ddev import-db` a normal db
`ddev import-db --no-drop` a smaller one with different tables. The union of the tables should now exist.
`ddev import-db --target-db=somedb` and verify that "somedb" now exists and has contents of the import.

## Automated Testing Overview:

* TestDdevImportDB was expanded to cover this territory
* TestCmdImportDB was expanded to handle import from named project
* TestCmdExportDB was added with basic test of ddev export-db
* TestDdevExportDB expanded

## Related Issue Link(s):

#1961 
#1652 

## Release/Deployment notes:
